### ISSUE TITLE
Allow access modifier indentation relative to start of line

### ISSUE THREAD

**Issue by AlexWayfer** [rubocop-hq/rubocop#5830]
Hello!

I can't fix this, sorry. But I found test:

```ruby
it 'registers an offense for misaligned private in module ' \
   'defined with Module.new' do
  inspect_source(<<-RUBY.strip_indent)
    Test = Module.new do

    private

      def test; end
    end
  RUBY
  expect(cop.offenses.size).to eq(1)
  expect(cop.messages)
    .to eq(['Indent access modifiers like `private`.'])
end
```

And I didn't found test for `no offenses` in this case.

So, I wrote it and it fails.

--------

## Expected behavior

Successful test:

```ruby
it 'accepts aligned private in module defined with Module.new' do
  expect_no_offenses(<<-RUBY.strip_indent)
    Test = Module.new do

      private

      def test; end
    end
  RUBY
end
```

## Actual behavior

Test fails. :(

## Steps to reproduce the problem

Add this case and run specs.

## RuboCop version

(a41dce832c3da1b31628fdc1d707a1254ac85ee2 from `master` branch)

```
$ ./bin/rubocop -V
0.55.0 (using Parser 2.5.1.0, running on ruby 2.5.1 x86_64-linux)
```

**Comment by Drenmi** [comment]
It checks indentation relative to the opening of its scope. I.e. this passes:

```ruby
Test = Module.new do

         private

         def test; end
       end

**Comment by AlexWayfer** [comment]
> I.e. this passes

And this is terrible for me.

**Comment by Drenmi** [comment]
> And this is terrible for me.

Fair enough. I relabelled the ticket as a feature request. 🙂

**Comment by AlexWayfer** [comment]
Another example:

```ruby
stub_const(
  'TestClassWithVeryLongName', Class.new(described_class) do
    # should I align body of class with class definition?
    # and why there are no offenses about indentation of methods?
  end
)
```

**Comment by AlexWayfer** [comment]
It looks fixed in RuboCop 0.66.
